dsp/src/lockin.rs

```diff
@@ -302,11 +263,13 @@ pub fn decimate(
 ///
 /// # Arguments
 ///
-/// * `signal` - Complex signal to decimate.
-pub fn magnitude_phase(signal: &mut [Complex<f32>]) {
+/// * `signal` - Complex signal for which the magnitude and phase
+/// should be computed. TODO currently, we compute the square of the
+/// magnitude. This should be changed to be the actual magnitude.
+pub fn magnitude_phase(signal: &mut [Complex<i32>]) {
     signal.iter_mut().for_each(|s| {
-        let new_i = libm::sqrtf([s.0, s.1].iter().map(|i| i * i).sum());
-        let new_q = libm::atan2f(s.1, s.0);
+        let new_i = [s.0, s.1].iter().map(|i| i * i).sum();
```

Comment:
This is longer and harder to understand than `s.0 * s.0 + s.1 * s.1`.

Reply:
Also again. We don't always need the magnitude output. Leave this to the user in `process()`.